devito/ir/clusters/algorithms.py

```diff
@@ -80,21 +80,26 @@ def guard(clusters):
     for c in clusters:
         # Find out what expressions in /c/ should be guarded
         mapper = {}
+        incond = []
         for e in c.exprs:
+            dimc = ()
             for k, v in e.ispace.sub_iterators.items():
                 for i in v:
                     if i.dim.is_Conditional:
-                        mapper.setdefault(i.dim, []).append(e)
-
+                        dimc += (i.dim,)
+            if len(dimc) > 0:
+                mapper.setdefault(dimc, []).append(e)
```

Comment:
@FabioLuporini I think this way is probably nicer. Mainly the mapper now looks like
{(xsub, ysub, tsub): expr}

This way the the guard is driven by the expression rather than the dimensions and take cares of all the associated conditions at once.